### PR TITLE
bbox library

### PR DESCRIPTION
**Pull request by ghost**
I am so sorry for the aux files. I tried to remove them by removing them in the directory and saying `git commit -m "removed aux file"` and indeed
	deleted:    bboxtest.aux
	deleted:    bboxtest.log
	deleted:    bboxtest.tex.bck
	deleted:    tikzlibrarybbox.code.tex.bck
but the files are still there. Is there anywhere a manual for git or do I have to continue guessing?

### CODE REVIEW

**Comment by hmenke** [comment]
This needs quite a bit of work.  I'll see whether I can do something over the weekend.

**Comment by hmenke** [comment]
A manual for git can be found on https://git-scm.com/

**Comment by ghost** [comment]
Hi Henri,

I agree that at this stage this is merely a proof of principle, and I did
not mean to dump that on you. If you give me a pointer to some similar code
from which I could infer which structures you prefer, I will be happy to
give it a shot.

File: experiments/Marmot/Venn/tikzlibraryvenn.code.tex
Comment (hmenke) [inline experiments/Marmot/Venn/tikzlibraryvenn.code.tex:18]: Uhm... What's the fundamental difference here?
https://github.com/pgf-tikz/pgf/blob/97a18d33fb1a791df41768f1709b6e477972b1e6/tex/generic/pgf/frontendlayer/tikz/tikz.code.tex#L535

File: experiments/Marmot/Venn/tikzlibraryvenn.code.tex
Comment (hmenke) [inline experiments/Marmot/Venn/tikzlibraryvenn.code.tex:26]: `current reverse clip` and `current reverse clip'` are the same thing or am I missing something?

File: experiments/Marmot/Venn/tikzlibraryvenn.code.tex
Comment (hmenke) [inline experiments/Marmot/Venn/tikzlibraryvenn.code.tex:29]: Why not just `Venn diagram/.style={venn/.cd,#1}`?

File: experiments/Marmot/Venn/tikzlibraryvenn.code.tex
Comment (hmenke) [inline experiments/Marmot/Venn/tikzlibraryvenn.code.tex:33]: “**Fewer** labels **than** Venn sets.”

File: experiments/Marmot/Venn/tikzlibraryvenn.code.tex
Comment (hmenke) [inline experiments/Marmot/Venn/tikzlibraryvenn.code.tex:61]: Is this where the enormous `execute at begin scope` ends?  I can't really see for the lack of indentation.  Also when doing this you have to somehow make sure that `Venn diagram` can only be used as a scope or picture option.

File: experiments/Marmot/3dtools/3dtoolstest.tex
Comment (hmenke) [inline experiments/Marmot/3dtools/3dtoolstest.tex:3]: Evidently this library is meant for interop with `tikz-3dplot`, which is why I won't merge this into mainline PGF/Ti*k*Z.  I'm not convinced by having more user macros.  I'm also not convinced by `\pgfmathtdparse`.  That seems very fragile.

File: experiments/Marmot/bbox/tikzlibrarybbox.code.tex
Comment (hmenke) [inline experiments/Marmot/bbox/tikzlibrarybbox.code.tex:1]: After some refactoring I have merged the `bbox` library into PGF/Ti*k*Z.  Thank you for your contribution.

File: experiments/Marmot/Venn/tikzlibraryvenn.code.tex
Comment (hmenke) [inline experiments/Marmot/Venn/tikzlibraryvenn.code.tex:1]: I wanted to merge the `venn` library, but the code is lacking in some places.  Would it also be possible to make it a PGF library rather than a pure Ti*k*Z library?

**Review by hmenke — CHANGES_REQUESTED** [review]

File: experiments/Marmot/3dtools/3dtoolstest.tex
Comment (hmenke) [reply experiments/Marmot/3dtools/3dtoolstest.tex:3]: In response to https://github.com/pgf-tikz/pgf/issues/731#issuecomment-553616722 which was probably meant to appear here:

---

The 3dtools library introduces a whole bunch of new user macros, like `\pgfmathtdparse`, `\pgfmathprintvector`, `\tdparse` (is that the same as `\pgfmathtdparse`?) and `\orthmat` (is this meant for users? If not, why is it public?).  This somewhat collides with the PGF/Ti*k*Z philosophy of using keys and values instead of commands.

Right now, the whole 3dtools library is in a single file `tikzlibrary3dtools.code.tex`.  It uses the LaTeX package `calculator` and `\newcommand`, i.e. it cannot be used with other formats.  Also it defines macros with `\pgf...` prefix which clearly belong in a PGF rather than a Ti*k*Z library.

Last but not least, I am totally not stopping you from publishing this library independently on CTAN for use with Ti*k*Z, just like `tikz-cd` or `tikz-feynman`.  So while I can understand your disappointment that this is probably not going to be in mainline without some vast improvement, I am slightly confused by the accusation of “destroying hours of work”.

Comment (hmenke) [reply experiments/Marmot/3dtools/3dtoolstest.tex:3]: I have packaged up `tikz-3dtools` and put it here: https://github.com/pgf-tikz/tikz-3dtools
Release v1.0 is also submitted to CTAN already.

Comment (hmenke) [reply experiments/Marmot/3dtools/3dtoolstest.tex:3]: `tikz-3dtools` has appeared on CTAN and should be in TeX Live soon.
https://ctan.org/pkg/tikz-3dtools

**Comment by ghost** [comment]
For some reason 3dtools is on CTAN

https://ctan.org/pkg/tikz-3dtools

All I can say is that I did not upload it, nor do I think it is in a stage
that allows it to be published. That is, I do not feel responsible for
complications.

**Comment by hmenke** [comment]
I'm getting more and more confused.  In https://github.com/pgf-tikz/pgf/issues/731#issuecomment-553616722 you seemed very worried about all the work you put into `tikz-3dtools` being in vain, so I went ahead, isolated the `3dtools` library from the rest of the PR and published it separately for your convenience.  It is entirely under your control, so if you want to have it taken down, just email ctan@ctan.org.

**Comment by ghost** [comment]
Whether or not the work in 3dtools is in vain will depend on whether or not
`\csname tikz@dcl@coord@#1\endcsname` will still work in the future. My
only concern was that it might get removed. It was not my concern that it
is not published. It should be only published if it is significantly more
advanced.

**Comment by hmenke** [comment]
The main part of this PR has been merged (the `bbox` library).  I'm therefore closing this for now.  You can open a new PR with your other changes.  Please rebase onto master before doing so.

**Pull request closed without merging**
